src/OnboardingSPA/components/Drawer/DrawerPanel/DesignColors.js

```diff
@@ -121,6 +266,100 @@ const DesignColors = () => {
 		}
 	}
 
+	// Pre-defined Colors Section
+
+	/**
+	 * Select a color from predefined palette and sync it.
+	 *
+	 * @param {string} colorStyle - Selected Color Palette slug
+	 */
+	const handleClick = ( colorStyle ) => {
+		if ( selectedColors?.slug === colorStyle ) {
+			return true;
+		}
+
+		clearCustomColors();
+		saveThemeColorPalette( colorStyle );
+		setSelectedColorsLocal( colorPalettes[ colorStyle ] );
+		LocalToState( colorPalettes[ colorStyle ], colorStyle );
+		trackHiiveEvent( 'color-selection', colorStyle );
+	};
+
+	/**
+	 * Build the predefined Color Palette Component
+	 *
+	 * @return {WPComponent} Predefined Color Palette Component
+	 */
+	function buildPredefinedPalette() {
+		return Object.keys( colorPalettes ).map( ( colorStyle, idx ) => {
+			return (
+				<div
+					key={ colorStyle }
+					className={ classNames(
+						'color-palette drawer-palette--button',
+						{
+							'color-palette-selected drawer-palette--button--selected':
+								colorStyle === currentData?.data?.colorStyle,
+						}
+					) }
+					role="button"
+					tabIndex={ idx + 1 }
+					onClick={ () => handleClick( colorStyle ) }
+					onKeyDown={ () => handleClick( colorStyle ) }
+				>
+					<div className="color-palette__colors">
+						<div
+							className="color-palette__colors--tertiary"
+							style={ {
+								backgroundColor: `${ colorPalettes[ colorStyle ]?.[ 'header-background' ] }`,
+							} }
+						/>
+						<div
+							className="color-palette__colors--secondary"
+							style={ {
+								backgroundColor: `${ colorPalettes[ colorStyle ]?.secondary }`,
+							} }
+						/>
+						<div
+							className="color-palette__colors--primary"
+							style={ {
+								backgroundColor: `${ colorPalettes[ colorStyle ]?.primary }`,
+							} }
+						/>
+					</div>
+					<div className="color-palette__name drawer-palette--button__text">
+						{ colorStyle?.charAt( 0 ).toUpperCase() +
+							colorStyle?.slice( 1 ) }
+					</div>
+				</div>
+			);
+		} );
+	}
+
+	// Custom Colors Section
+
+	/**
+	 * Checks id custom colors has a value
+	 * used to toggle the visiblity of Reset button
+	 *
+	 * @return {boolean} isCustomColorActive
+	 */
+	function isCustomColorActive() {
+		for ( const custom in customColors ) {
+			if ( customColors[ custom ] !== '' ) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Find the index of the color type being changed in the array
+	 * Array to check -> [{color: '#ffffff', name: 'Base', slug: 'base'}, ...]
+	 *
+	 * @param {string} slugName - Slug Name for color i.e. base, secondary, header-title,...
+	 * @return {number} res - Index of the key "slug" mapped ({color: '#ffffff', name: 'Base', slug: 'base'}, ...) in the array
+	 */
 	function findInCustomColors( slugName ) {
```

Comment:
So basically if a wrong mapping is done in backend say `base -> 'headwer-title'` then it will search for index of **'headwer-title'** in palette but when that won't be there it will return the index of '**base**'